backend/de.metas.async/src/main/java/de/metas/async/service/AsyncBatchService.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * #%L
+ * de.metas.async
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.async.service;
+
+import ch.qos.logback.classic.Level;
+import de.metas.async.AsyncBatchId;
+import de.metas.async.api.IAsyncBatchBL;
+import de.metas.async.api.IAsyncBatchDAO;
+import de.metas.async.eventbus.AsyncBatchEventBusService;
+import de.metas.async.eventbus.AsyncBatchNotifyRequest;
+import de.metas.async.model.I_C_Async_Batch;
+import de.metas.async.model.I_C_Queue_WorkPackage;
+import de.metas.async.model.validator.C_Queue_WorkPackage;
+import de.metas.logging.LogManager;
+import de.metas.util.Loggables;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.compiere.util.Env;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.function.Supplier;
+
+@Service
+public class AsyncBatchService
+{
+	private static final Logger logger = LogManager.getLogger(AsyncBatchService.class);
+
+	private final IAsyncBatchDAO asyncBatchDAO = Services.get(IAsyncBatchDAO.class);
+	private final IAsyncBatchBL asyncBatchBL = Services.get(IAsyncBatchBL.class);
+
+	private final AsyncBatchObserver asyncBatchObserver;
+	private final AsyncBatchEventBusService asyncBatchEventBusService;
+
+	public AsyncBatchService(
+			@NonNull final AsyncBatchObserver asyncBatchObserver,
+			@NonNull final AsyncBatchEventBusService asyncBatchEventBusService)
+	{
+		this.asyncBatchObserver = asyncBatchObserver;
+		this.asyncBatchEventBusService = asyncBatchEventBusService;
+	}
+
+	public void checkProcessed(@NonNull final AsyncBatchId asyncBatchId, @Nullable final String trxName)
```

Comment:
The suggestion to have in place the parent-child logic was to have traceability when a batch is created inside of another batch execution.

In the current scenarios we don't really need to wait for the child async batches, as that part is already taken care of by the orchestration code. But, I see your point, also consulted with @cp-ps. It might be useful to have that for future tasks.. but if we do this check we need to replace `de.metas.async.api.impl.AsyncBatchBL#checkProcessed` a.k.a. the old way of setting the `isProcessed` flag with what we are currently doing when notifying that an async batch is finished i.e. `de.metas.async.service.AsyncBatchService#checkProcessed`